spec.mediawiki

```diff
@@ -978,239 +1196,318 @@ for verification and relay to the counterparty.
 | timestamp || 8  || the client's UNIX timestamp
 |}
 
-==Concept of Operations==
+===Match revocation===
+
+A match can be revoked by the server if a client fails to act within the
+[[#Exchange_Variables|broadcast timeout]]. A match revocation will result in
+penalties for the violating party only.
+The revoked match quantity is not added back to the orderbook in any form.
+
+'''JSON-RPC method:''' <code>revoke_match</code>, '''originator:''' DEX
+
+{|
+! field    !! type   !! description
+|-
+| orderid  || string || order ID
+|-
+| matchid  ||  int  || the match ID
+|-
+| sig      || string || server hex-encoded signature of serialized revocation. serialization described below
+|}
 
-This section describes how the exchange service matches orders from registered 
-users, and facilitates order execution by the clients directly on their assets' 
-blockchains.
+'''Revocation serialization'''
 
-===Account Creation===
+{|
+! field      !! size (bytes)  !! description
+|-
+| orderid    || 32  || the order ID
+|-
+| matchid    || 8  || the ID assigned to this match
+|}
 
-Access to markets is granted to client identities represented in long-lived PKI 
-key pairs [ref?] fee to the exchange.
-The process of establishing an identity on the server and gaining authorization 
-for a particular market proceeds as follows.
+==Trade Suspension==
 
-# Client connects to server and both identify themselves using a long-lived PKI keypair after the associated client has paid a (one-time) fee.
+There are a number of scenarios where the server may suspend operations,
+intentionally or not.
+During trade suspension, standing limit orders are not revoked. If the server
+intentionally suspends operations, they should provide a notification to
+connected clients as early as possible, ideally with several epochs for the
+client to get their orders situated before matching ceases.
+The epoch specified in the suspension notification is the first epoch for which
+orders will no longer be accepted. Users should expect to lose connection
+during suspension. When possible, the server should settle all existing matches
+before closing connections. If that is not possible, clients will need to
+reconnect and complete settlement when the server comes back online.
+If the server disconnects without notice, it is expected that orders placed
+during the current epoch are revoked at no penalty to the client.
 
-# Server responds with a fee that the client must pay to get a trading account on the server.
+Standing limit order are only revoked (no penalty) if the client does not
+reconnect before the next [[#Session_Authentication|startepoch]].
```

Comment:
There will need to be at least 1 full epoch after the server comes back online to let clients reconnect.  Ideally there would be N epochs after the server comes back online for which no orders are accepted, when clients may reestablish their connections.

Reply:
Added a `startepoch` parameter to a `connect` response.